Tests/test_dependencies.py

```diff
@@ -0,0 +1,200 @@
+import json
+import math
+
+
+class TestVertex:
+    def __init__(self, test_name):
+        self.neighbors = {}
+        self.test_name = test_name
+
+        self.visited = False
+
+    def add_neighbor(self, neighbor_test):
+        neighbor_name = neighbor_test.test_name
+        self.neighbors[neighbor_name] = neighbor_test
+
+    def get_connected_component(self, tests_in_component):
+        tests_in_component.append(self.test_name)
+        self.visited = True
+        for neighbor_name in self.neighbors:
+            neighbor_vertex = self.neighbors[neighbor_name]
+            if not neighbor_vertex.visited:
+                tests_in_component = neighbor_vertex.get_connected_component(tests_in_component)
+        return tests_in_component
+
+
+class TestsGraph:
+    def __init__(self):
+        self.test_vertices = {}
+        self.clusters = []
+
+    def add_test_graph_vertices(self, tests_data):
+        for test_playbook_record in tests_data:
+            playbook_name_in_record = test_playbook_record.get("playbookID")
+            if playbook_name_in_record and playbook_name_in_record not in self.test_vertices:
+                new_test_vertex = TestVertex(playbook_name_in_record)
+                self.test_vertices[playbook_name_in_record] = new_test_vertex
+
+    def add_test_graph_neighbors(self, tests_data):
+        integration_to_tests_mapping = get_integration_to_tests_mapping(tests_data)
+        for integration_name in integration_to_tests_mapping:
+            tests_using_integration = integration_to_tests_mapping[integration_name]
+            for i in range(len(tests_using_integration)):
+                first_test_name = tests_using_integration[i]
+                first_test_vertex = self.test_vertices[first_test_name]
+
+                for j in range(i + 1, len(tests_using_integration)):
+                    second_test_name = tests_using_integration[j]
+                    second_test_vertex = self.test_vertices[second_test_name]
+
+                    first_test_vertex.add_neighbor(second_test_vertex)
+                    second_test_vertex.add_neighbor(first_test_vertex)
+
+    def get_clusters(self):
+        clusters = []
+        for test_name in self.test_vertices:
+            test_vertex = self.test_vertices[test_name]
+            if not test_vertex.visited:
+                test_connected_component = test_vertex.get_connected_component([])
+                clusters.append(test_connected_component)
+        self.clusters = clusters
+
+    def build_tests_graph_from_conf_json(self, tests_file_path):
```

Comment:
you should receive `dependent_tests` you already have it when you launch this func